### PR TITLE
To support the remarketing tag for GA Premium

### PR DESCRIPTION
**Pull request by CarolineGeven**
Support premium remarketing tag.

When merging, please also merge this one: https://github.com/Yoast/google-analytics-premium/pull/103

### CODE REVIEW

File: frontend/class-universal.php [hunk 1]
```diff
@@ -82,6 +82,15 @@ public function tracking( $return_array = false ) {
 				$gaq_push[] = "'set', 'anonymizeIp', true";
 			}
 
+			/**
+			* Filter: 'yst_ga_filter_push_vars' - Allow changing the $gaq_push variables before scripts are required.
+			*
+			* @api array
+			*/
+			if ( has_filter( 'yst_ga_filter_push_vars' ) && $value_to_push = apply_filters( 'yst_ga_filter_push_vars', $gaq_push ) ) {
+					$gaq_push[] = $value_to_push;
```
Comment (andizer) [inline frontend/class-universal.php]: Indentions is weird

File: admin/pages/settings.php [hunk 2]
```diff
@@ -101,6 +101,8 @@
 		echo Yoast_GA_Admin_Form::input( 'checkbox', __( 'Anonymize IPs', 'google-analytics-for-wordpress' ), 'anonymize_ips', null, sprintf( __( 'This adds %1$s, telling Google Analytics to anonymize the information sent by the tracker objects by removing the last octet of the IP address prior to its storage.', 'google-analytics-for-wordpress' ), '<a href="https://developers.google.com/analytics/devguides/collection/gajs/methods/gaJSApi_gat?csw=1#_gat._anonymizeIp" target="_blank"><code>_anonymizeIp</code></a>' ) );
 		echo Yoast_GA_Admin_Form::select( __( 'Ignore users', 'google-analytics-for-wordpress' ), 'ignore_users', $yoast_ga_admin->get_userroles(), __( 'Users of the role you select will be ignored, so if you select Editor, all Editors will be ignored.', 'google-analytics-for-wordpress' ), true );
 		echo Yoast_GA_Admin_Form::input( 'checkbox', __( 'Disable analytics dashboard', 'google-analytics-for-wordpress' ), 'dashboards_disabled', null, __( 'This will completely disable the dashboard and stop the plugin from fetching the latest analytics data.', 'google-analytics-for-wordpress' ) );
+
+		do_action( 'yst_ga-universal-tab' );
```
Comment (jdevalk) [inline admin/pages/settings.php]: This lacks documentation.

**Comment by jdevalk** [comment]
CR 1: done.

**Comment by atimmer** [comment]
We should consistently use underscores or dashes in actions/filters. `yst_ga-universal-tab` is not consistent with `yst_ga_filter_push_vars`. It isn't perfect in the current codebase but it should be in new code.

WordPress core uses underscores consistently, so we should as well.

File: admin/pages/settings.php [hunk 3]
```diff
@@ -101,6 +101,11 @@
...
 		echo Yoast_GA_Admin_Form::select( __( 'Ignore users', 'google-analytics-for-wordpress' ), 'ignore_users', $yoast_ga_admin->get_userroles(), __( 'Users of the role you select will be ignored, so if you select Editor, all Editors will be ignored.', 'google-analytics-for-wordpress' ), true );
 		echo Yoast_GA_Admin_Form::input( 'checkbox', __( 'Disable analytics dashboard', 'google-analytics-for-wordpress' ), 'dashboards_disabled', null, __( 'This will completely disable the dashboard and stop the plugin from fetching the latest analytics data.', 'google-analytics-for-wordpress' ) );
+
+		/**
+		 * Action: 'yst_ga_universal_tab' - Allow changing the universal tab of the settings
```
Comment (atimmer) [inline admin/pages/settings.php]: The action does not allow changing the tab, it only allows adding new stuff. So it should be changed to `Allow adding to the universal (...)`

File: admin/pages/settings.php [hunk 4]
```diff
@@ -127,6 +132,9 @@
 		echo Yoast_GA_Admin_Form::input( 'checkbox', __( 'Add <code>_setAllowLinker</code>', 'google-analytics-for-wordpress' ), 'add_allow_linker', null, sprintf( __( 'This adds a %1$s call to your tracking code, allowing you to use %2$s and related functions.', 'google-analytics-for-wordpress' ), '<a href="https://developers.google.com/analytics/devguides/collection/gajs/methods/gaJSApiDomainDirectory?csw=1#_gat.GA_Tracker_._setAllowLinker" target="_blank"><code>_setAllowLinker</code></a>', ' <code>_link</code>' ) );
 		echo Yoast_GA_Admin_Form::textarea( __( 'Custom code', 'google-analytics-for-wordpress' ), 'custom_code', sprintf( __( 'Not for the average user: this allows you to add a line of code, to be added before the %1$s call.', 'google-analytics-for-wordpress' ), '<a href="https://developers.google.com/analytics/devguides/collection/gajs/methods/gaJSApiBasicConfiguration#_gat.GA_Tracker_._trackPageview" target="_blank"><code>_trackPageview</code></a>' ) );
 
+		/**
+		 * Action: 'yst_ga_advanced-tab' - Allow changing the advanced tab of the settings
```
Comment (atimmer) [inline admin/pages/settings.php]: The action does not allow changing the tab, it only allows adding new stuff. So it should be changed to `Allow adding to the advanced (...)`

File: admin/pages/settings.php [hunk 5]
```diff
@@ -127,6 +132,9 @@
...
+		 */
 		do_action( 'yst_ga_advanced-tab' );
```
Comment (atimmer) [inline admin/pages/settings.php]: We decided on using underscores, so we should stick to that.

Comment (CarolineGeven) [inline admin/pages/settings.php]: Unfortunately we can't with this action, as it's an old one. This was discussed with @omarreiss yesterday. We can only change new actions/filters to underscores, not old ones.

**Comment by atimmer** [comment]
CR Done.

**Pull request merged**